velox/exec/RowContainer.h

```diff
@@ -127,10 +131,23 @@ class RowContainer {
   // Allocates a new row and initializes possible aggregates to null.
   char* newRow();
 
+  uint32_t rowSize(const char* row) const {
+    return fixedRowSize_ +
+        (rowSizeOffset_
+             ? *reinterpret_cast<const uint32_t*>(row + rowSizeOffset_)
+             : 0);
+  }
+
   // Adds 'rows' to the free rows list and frees any associated
   // variable length data.
   void eraseRows(folly::Range<char**> rows);
 
+  void incrementRowSize(char* row, uint64_t bytes) {
```

Comment:
This logic is duplicated between Aggregate and RowContainer. Is there any particular reason group by doesn't use RowContainer?

Reply:
There are aggregates that are not in a RowContainer for global agg but might as well be in one.
More importantly, the aggregates do not knowabout RowContainer, onlly about HashStringAllocator. So duplicating this is more encapsulation than making RowContainer and Aggregtes know about each other.